web/src/app/admin/AdminAPIKeys.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState } from 'react'
+import makeStyles from '@mui/styles/makeStyles'
+import {
+  Button,
+  Grid,
+  Typography,
+  Card,
+  ButtonBase,
+  CardHeader,
+} from '@mui/material'
+import { Add } from '@mui/icons-material'
+import AdminAPIKeysDrawer from './admin-api-keys/AdminAPIKeysDrawer'
+import { GQLAPIKey, CreatedGQLAPIKey } from '../../schema'
+import { Time } from '../util/Time'
+import { gql, useQuery } from 'urql'
+import FlatList, { FlatListListItem } from '../lists/FlatList'
+import Spinner from '../loading/components/Spinner'
+import { GenericError } from '../error-pages'
+import { Theme } from '@mui/material/styles'
+import AdminAPIKeysActionDialog from './admin-api-keys/AdminAPIKeysActionDialog'
+import AdminAPIKeysTokenDialog from './admin-api-keys/AdminAPIKeysTokenDialog'
+import { DateTime } from 'luxon'
+// query for getting existing API Keys
+const query = gql`
+  query gqlAPIKeysQuery {
+    gqlAPIKeys {
+      id
+      name
+      description
+      createdAt
+      createdBy {
+        id
+        name
+      }
+      updatedAt
+      updatedBy {
+        id
+        name
+      }
+      lastUsed {
+        time
+        ua
+        ip
+      }
+      expiresAt
+      allowedFields
+      role
+    }
+  }
+`
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    '& .MuiListItem-root': {
+      'border-bottom': '1px solid #333333',
+    },
+  },
+  buttons: {
+    'margin-bottom': '15px',
+  },
+  containerDefault: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '100%',
+      transition: `max-width ${theme.transitions.duration.leavingScreen}ms ease`,
+    },
+    '& .MuiListItem-root': {
+      padding: '0px',
+    },
+  },
+  containerSelected: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '70%',
+      transition: `max-width ${theme.transitions.duration.enteringScreen}ms ease`,
+    },
+    '& .MuiListItem-root': {
+      padding: '0px',
+    },
+  },
+}))
+
+export default function AdminAPIKeys(): JSX.Element {
+  const classes = useStyles()
+  const [selectedAPIKey, setSelectedAPIKey] = useState<GQLAPIKey | null>(null)
+  const [tokenDialogClose, onTokenDialogClose] = useState(false)
+  const [openActionAPIKeyDialog, setOpenActionAPIKeyDialog] = useState(false)
+  const [create, setCreate] = useState(false)
+  const emptyAPIKey = {
+    id: '',
+    name: '',
+    description: '',
+    createdAt: '',
+    createdBy: null,
+    updatedAt: '',
+    updatedBy: null,
+    lastUsed: null,
+    expiresAt: DateTime.utc().plus({ days: 7 }).toISO(),
+    allowedFields: [],
+    role: 'user',
+  }
```

Comment:
Let's move this out of the list and have the dialogs own the data:
- Split ActionDialog into Create and Edit dialogs
- Instead of TokenDialog, just have it be part of the create dialog's code to display the token at the end
- EditDialog should just take the key ID (similar to the Delete dialog)

Reply:
Noted. The items has been implemented.